hazelcast/src/main/java/com/hazelcast/sql/impl/SqlServiceImpl.java

```diff
@@ -16,52 +16,79 @@
 import com.hazelcast.internal.serialization.InternalSerializationService;
+import com.hazelcast.internal.util.Preconditions;
 import com.hazelcast.logging.ILogger;
 import com.hazelcast.spi.impl.NodeEngine;
 import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.sql.SqlResult;
+import com.hazelcast.sql.SqlQuery;
+import com.hazelcast.sql.SqlService;
 import com.hazelcast.sql.impl.optimizer.DisabledSqlOptimizer;
 import com.hazelcast.sql.impl.optimizer.OptimizationTask;
 import com.hazelcast.sql.impl.optimizer.SqlOptimizer;
 import com.hazelcast.sql.impl.optimizer.SqlPlan;
 import com.hazelcast.sql.impl.plan.Plan;
 import com.hazelcast.sql.impl.state.QueryState;
 
+import javax.annotation.Nonnull;
 import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.function.Consumer;
 import java.util.logging.Level;
 
 /**
  * Base SQL service implementation that bridges optimizer implementation, public and private APIs.
  */
-public class SqlServiceImpl implements Consumer<Packet> {
+public class SqlServiceImpl implements SqlService, Consumer<Packet> {
     /** Outbox batch size in bytes. */
     private static final int OUTBOX_BATCH_SIZE = 512 * 1024;
 
     /** Default state check frequency. */
-    private static final long STATE_CHECK_FREQUENCY = 10_000L;
+    private static final long STATE_CHECK_FREQUENCY = 1_000L;
 
     private static final String OPTIMIZER_CLASS_PROPERTY_NAME = "hazelcast.sql.optimizerClass";
     private static final String SQL_MODULE_OPTIMIZER_CLASS = "com.hazelcast.sql.impl.calcite.CalciteSqlOptimizer";
 
     private SqlOptimizer optimizer;
     private final ILogger logger;
     private final NodeEngineImpl nodeEngine;
+    private final long queryTimeout;
+
+    private final NodeServiceProviderImpl nodeServiceProvider;
 
     private volatile SqlInternalService internalService;
 
     public SqlServiceImpl(NodeEngineImpl nodeEngine) {
         this.nodeEngine = nodeEngine;
         logger = nodeEngine.getLogger(getClass());
 
-        // These two parameters will be taken from the config, when public API is ready.
-        int operationThreadCount = Runtime.getRuntime().availableProcessors();
-        int fragmentThreadCount = Runtime.getRuntime().availableProcessors();
+        SqlConfig config = nodeEngine.getConfig().getSqlConfig();
+
+        int executorPoolSize = config.getExecutorPoolSize();
+        int operationPoolSize = config.getOperationPoolSize();
+        long queryTimeout = config.getQueryTimeoutMillis();
 
-        NodeServiceProviderImpl nodeServiceProvider = new NodeServiceProviderImpl(nodeEngine);
+        if (executorPoolSize == SqlConfig.DEFAULT_EXECUTOR_POOL_SIZE) {
+            executorPoolSize = Runtime.getRuntime().availableProcessors();
+        }
+
+        if (operationPoolSize == SqlConfig.DEFAULT_OPERATION_POOL_SIZE) {
+            operationPoolSize = Runtime.getRuntime().availableProcessors();
```

Comment:
This used to be `Math.min(8, Runtime.getRuntime().availableProcessors())`, was the change intentional?

Reply:
Math.min was very old heuristics with no proven value. It is simplified now.